src/comps/profile/comps/ViewPurchases.jsx

```diff
@@ -6,9 +6,11 @@ export default function ViewPurchases() {
   const valContext = useContext(contextApi);
   return (
     <div>
-    {valContext.userData.historyOfCart.length!==0?valContext.userData.historyOfCart.map((valProduct) => {
-      return <Product valProduct={valProduct} />
-  }):'No Purchases'}
+      {valContext.userData.historyOfCart.length !== 0
+        ? valContext.userData.historyOfCart.map((valProduct) => {
+            return <Product valProduct={valProduct} />;
+          })
+        : "No Purchases"}
```

Comment:
You can save `valContext.userData.historyOfCart` into a variable to make your code a bit cleaner.
Also extract No purchases to constant file